grpc-proxy/src/main/common/docker-entrypoint.sh

```diff
@@ -2,10 +2,38 @@
 
 set -o nounset
 
-exec /app/dist/grpcwebproxy \
-  --run_tls_server=false \
-  --backend_addr="${BACKEND_ADDR}" \
-  --backend_tls=false \
-  --allow_all_origins \
-  --use_websockets \
-  --backend_max_call_recv_msg_size=104857600
+TLS_DIR="${TLS_DIR:-/etc/deephaven/certs}"
+TLS_CRT="${TLS_CRT:-$TLS_DIR/tls.crt}"
+TLS_KEY="${TLS_KEY:-$TLS_DIR/tls.key}"
+TLS_CA="${TLS_CA:-$TLS_DIR/ca.crt}"
+
+PROXY_TLS_PORT="${PROXY_TLS_PORT:-8443}"
+PROXY_DEBUG_PORT="${PROXY_DEBUG_PORT:-8080}"
+
+ALLOWED_ORIGINS="${ALLOWED_ORIGINS:+--allowed_origins=$ALLOWED_ORIGINS}"
+ALLOWED_ORIGINS="${ALLOWED_ORIGINS:---allow_all_origins}"
+
+if [ -s "$TLS_CRT" ]; then
+
+    exec /app/dist/grpcwebproxy \
+      --run_http_server=false \
+      --server_http_debug_port="${PROXY_DEBUG_PORT}" \
+      --backend_addr="${BACKEND_ADDR}" \
+      --server_http_tls_port="${PROXY_TLS_PORT}" \
+      --server_tls_cert_file="$TLS_CRT" \
+      --server_tls_key_file="$TLS_KEY" \
+      --server_tls_client_ca_files="$TLS_CA,/etc/ssl/certs/ca-certificates.crt,/var/run/secrets/kubernetes.io/serviceaccount/ca.crt" \
```

Comment:
Is this kubernetes path going to break anybody who may want to do TLS on non-kube?

Reply:
Nope; I don't _think_ so at least. It's just a list of places where there are CA files.

If you want, I can build the list and check if the files are readable first, that way it will work in kube but won't risk breaking someone w/o that ca cert.

Reply:
I checked if they exist before adding to the flag, so there's no way it can break non-kube users now.